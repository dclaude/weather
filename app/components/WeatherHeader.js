import React from 'react'
import PropTypes from 'prop-types'

const iconNames = [ '01d', '01n', '02d', '02n', '03d', '03n', '04d', '04n', '09d', '09n', '10d', '10n', '11d', '11n', '13d', '13n', '50d', '50n' ]
const icons = iconNames.reduce((acc, icon) => {
  acc[icon] = require(`../images/weather-icons/${icon}.svg`)
  return acc
}, {})

const WeatherHeader = ({ icon, dt }) => {
  const date = new Date(0)
  date.setUTCSeconds(dt)
  const options = { weekday: 'long', month: 'long', day: 'numeric' }
  return (
    <div className='WeatherHeader-container'>
      <img className='WeatherHeader-img' src={`${icons[icon]}`} alt='Weather icon' />
      <h2 className='WeatherHeader-header'>{date.toLocaleDateString(undefined, options)}</h2>
    </div>
  )
}

WeatherHeader.propTypes = {
  icon: PropTypes.string.isRequired,
  dt: PropTypes.number.isRequired,
}

export default WeatherHeader

