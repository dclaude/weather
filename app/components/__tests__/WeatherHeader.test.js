import React from 'react'
import WeatherHeader from '../WeatherHeader'
import renderer from 'react-test-renderer'
import { shallow } from 'enzyme'
import ReactDOM from 'react-dom'

test('test with react-dom', () => {
  const node = document.createElement('div');
  ReactDOM.render(<WeatherHeader icon='01n' dt={1521975600} />, node)
  expect(node.innerHTML).toContain('alt="Weather icon"');
})

test('test with react-test-renderer', () => {
  const component = renderer.create(
    <WeatherHeader icon='01n' dt={1521975600} />
  )
  expect(component.toJSON()).toMatchSnapshot()
})

test('test with enzyme', () => {
  const comp = shallow(<WeatherHeader icon='01n' dt={1521975600} />);
  expect(comp.find('h2').text()).toEqual('M03 25, Sun')
  const img = comp.find('img')
})

