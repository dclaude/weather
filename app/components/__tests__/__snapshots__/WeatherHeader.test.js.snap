// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test with react-test-renderer 1`] = `
<div
  className="WeatherHeader-container"
>
  <img
    alt="Weather icon"
    className="WeatherHeader-img"
    src="test-file-stub"
  />
  <h2
    className="WeatherHeader-header"
  >
    M03 25, Sun
  </h2>
</div>
`;
